### DISCUSSION TITLE
can I call the Radio module in a Pico w ?

### DISCUSSION THREAD

**Discussion by melvincramer911** [micropython/micropython#10334] [RP2040 / Pico]
I would like to communicate between pico w computers and make them the systems not dependent on my home network.   Stated another way, If I am at an friends house and want to make this work, I do not need that person's wifi.  

I was using the openai chat bot and the system told me that I can run this on 2 pico w systems but I can not find the radio module: 

import radio

# Initialize the radio
radio.on()

# Set the radio to transmit mode
radio.config(mode=radio.MODE_TX)

# Send a message
radio.send("Hello, World!")

import radio

# Initialize the radio
radio.on()

# Set the radio to receive mode
radio.config(mode=radio.MODE_RX)

# Wait for a message
msg = radio.receive()
print(msg)

**Comment by dhalbert** [comment]
ChatGPT doesn't know what it's talking about. It looks like it's cribbing off the `radio` module for micro:bit.

Don't trust ChatGPT as a source of knowledge. It could be right, or it could be completely wrong, and you can't tell. It's just cobbling together phrases from a large language model. It's like asking a parrot with a very large memory a question. Use a search engine.

**Comment by dhalbert** [comment]
You _can_ set up one Pico W as an access point and have another one get on the network that the access point presents. Then you can use whatever you want to connect between the two, such as HTTP. E.g. https://www.recantha.co.uk/blog/?p=21398, and there are many other examples.

**Comment by melvincramer911** [comment]
Thank you very much Dan. Can I assign static ip addresses and not use the wifi hub at all Name of machines:Mighty01  192.161.10.1Minion10  192.161.10.10Minion11  192.161.10.11Minion12  192.161.10.12All Minion* systems could talk to Mighty01.  Mighty01 would have a display and keyboard.  I may upgrade the Mighty01 to a Pi zero 2 w :)Having Mighty01 handing out ip addresses would make a lot of not required complexity What are you doing helping us out on Christmas Break ?   ;)  Tank you > -------- Original Message --------> Subject: Re: [micropython/micropython] can I call the Radio module in a> Pico w ? (Discussion #10334)> From: Dan Halbert ***@***.***>> Date: Mon, December 26, 2022 5:22 pm> To: micropython/micropython ***@***.***>> Cc: melvincramer911 ***@***.***>,  Author> ***@***.***>> > > You _can_ set up one Pico W as an access point and have another one get on the network that the access point presents. Then you can use whatever you want to connect between the two, such as HTTP. E.g. https://www.recantha.co.uk/blog/?p=21398, and there are many other examples.> > -- > Reply to this email directly or view it on GitHub:> https://github.com/orgs/micropython/discussions/10334#discussioncomment-4499546> You are receiving this because you authored the thread.> > Message ID: ***@***.***>

**Reply by melvincramer911** [reply]
that sure came out as a mess. sorry Dan.  

does this look like a good example of what I need if I want static ip addresses ? 

https://forum.core-electronics.com.au/t/raspberry-pi-pico-w-access-point-cannot-change-default-ip/15039

Thank you Dan

**Comment by dhalbert** [comment]
@melvincramer911 Sorry, someone with more knowledge of the Pico W API could answer. I am much more familiar with the CircuitPython WiFi API for Pico W.

**Comment by peterhinch** [comment]
The range of WiFi signals is too short to enable this to work unless your friend lives extremely close. 1-200 metres max without special antennas.

**Comment by melvincramer911** [comment]
Hello

Thank you for your reply.

Curious, can circuit  python do radio communications on the picow
without an ip address ?   I would love to not need the to add a nrf24l01
the pico.

Regards

David

**Reply by dhalbert** [reply]
The `radio` module you mentioned in the original post uses simple communication over BLE, which is not yet supported on the Pico W. CircuitPython and MicroPython have similar WiFi support on the Pico W: they are both based on a single lower-level library.
